### PR TITLE
Added rating to performers and studios

### LINKED ISSUE #1283: [Feature] Star Ratings for Performers
**Is your feature request related to a problem? Please describe.**
Performer tags are great, and have made filtering performers so much easier. There’s still no way to surface the ones you like though, aside from favorites. Unfortunately that puts all favorites on an equal level, which they aren’t. 

**Describe the solution you'd like**
Add a rating field to the performer page, implemented in the same way as scenes, images, and galleries.

**Describe alternatives you've considered**
1. You can instead have the ratings roll up, meaning that the performer’s star rating is an average of all their rated media.
2. You can create “5-star” performer tags. This is what I’m trying now, but you don’t have any way to sort by them efficiently.

**Additional context**
As someone with ADHD, if a performer is out of sight, then they’re out of mind. When I find someone I prefer, I would like to be able to assign the star rating so they can be easily resurfaced later via sorting. Plus when I get bored of the 5 star performers, I can easily scroll down to the 4 stars, then the 3 stars, etc.

### LINKED ISSUE #1289: [Feature] Studio Star Ratings
**Is your feature request related to a problem? Please describe.**
Similar to #1283 this would solve the issue of it being hard to surface studios that you prefer. Being able to assign star ratings would allow you to not only rate each studio, but to sort them in order of rating.

**Describe the solution you'd like**
Add a rating field to the studio page, implemented in the same way as scenes, images, and galleries.

**Describe alternatives you've considered**
You can instead have the ratings roll up, meaning that the studio’s star rating is an average of all their rated media.

**Additional context**
As someone with ADHD, if a studio is out of sight, then they’re out of mind. When I find a studio I prefer, I would like to be able to assign a star rating so they can be easily resurfaced later via sorting. Plus when I get bored of the 5 star studios, I can easily scroll down to the 4 stars, then the 3 stars, etc.

### PR DESCRIPTION
**Pull request by julien0221**
I added rating on performers and studios page. And I added some tests to it.

Resolves #1283 
Resolves #1289

### CODE REVIEW

File: ui/v2.5/src/components/Performers/PerformerDetails/PerformerEditPanel.tsx [hunk 1]
```diff
@@ -62,7 +63,7 @@ export const PerformerEditPanel: React.FC<IPerformerDetails> = ({
   // Editing state
   const [scraper, setScraper] = useState<GQL.Scraper | undefined>();
   const [newTags, setNewTags] = useState<GQL.ScrapedSceneTag[]>();
-
+  const [rating, setRating] = useState<number>(performer?.rating ?? NaN);
```
Comment (WithoutPants) [inline ui/v2.5/src/components/Performers/PerformerDetails/PerformerEditPanel.tsx:66]: Should be using formik values instead.

This use of this means that if you navigate away from the edit page after changing the rating but not saving, then no prompt is displayed about unsaved data.

**Review by WithoutPants — CHANGES_REQUESTED** [review]
* setting a rating on a Performer that already is marked as a favourite results in overlapping ribbons in the Performers page:
image

My suggestion would be to change the favourite ribbon to be a heart icon on the upper right of the performer image.

* rating is not set when creating a performer

**Comment by julien0221** [comment]
I removed the banner favourite and I added a heart icon on upper right as you suggested.
I changed the setRating to use formik to show the unsaved popup.

I created some performers and I am not sure what you mean with "rating is not set when creating a performer". Because it works all good when I was testing.
Can you give me more details please?

**Review by WithoutPants — APPROVED** [review]
The create bug was fixed with the change to use formik values.

**Comment by WithoutPants** [comment]
@julien0221 please contact me on Discord to arrange bounty payment.

**Comment by GsMumbo** [comment]
@julien0221 @WithoutPants It looks like Star Ratings were added to both, but only Performers can be sorted by rating. We’re Studios also supposed to have this sorting too?

(Also thank you! This is amazing!)

**Pull request merged**